packages/compass-components/src/components/drawer/drawer-toolbar-layout/drawer-toolbar-layout-container.tsx

```diff
@@ -39,9 +39,13 @@ export const DrawerToolbarLayoutContainer = forwardRef<
   ) => {
     const { openDrawer, closeDrawer, getActiveDrawerContent, isDrawerOpen } =
       useDrawerToolbarContext();
-    const { id, title, content } = getActiveDrawerContent() || {};
+    const { id } = getActiveDrawerContent() || {};
     const lgIds = getLgIds(dataLgId);
     const hasData = toolbarData && toolbarData.length > 0;
+    const { title, content } =
+      toolbarData.find((data) => {
+        return data.id === id;
+      }) ?? {};
```

Comment:
This is like a super minimal version of the fix that LG team did in [this patch](https://github.com/mongodb/leafygreen-ui/pull/3016), but we can be sure that when this is not vendored anymore, this will continue working

Reply:
Added in https://github.com/mongodb-js/compass/pull/7176/commits/1538798844d9bdc40aacbccd395c97a9a1785d2e